include/ethernet.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (C) 2020  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_ETHERNET_H
+#define DOSBOX_ETHERNET_H
+
+#include "config.h"
+#include <functional>
+
+/** A virtual Ethernet connection
+ * While emulated Ethernet adapters provide the ability for the guest OS to
+ * send and receive Ethernet packets, the emulator itself needs to pass these
+ * packets to device on the host system.
+ * The EthernetConnection class provides a virtual connection for passing
+ * packets to and from a backend, intended for use by emulated adapter code.
+ */
+class EthernetConnection
+{
+    public:
+        /** Initializes the connection.
+         * After creating an EthernetConnection, it must have its connection
+         * opened and initialized before further use.
+         * This is done in a separate function instead of a constructor so any
+         * error code may be propagated.
+         * Do not call this function twice.
+         * @return True if the connection is initialized, false otherwise
+         */
+        virtual bool Initialize(void) = 0;
+
+        /** Closes the connection.
+         * Owners of an EthernetConnection are expected to delete it when
+         * finished with the connection.
+         * The destructor will close the connection and free any
+         * per-connection resources.
+         */
+        virtual ~EthernetConnection(void)
```

Comment:
Suggest snipping `(void)` to simply `()`, as it's just as understandable for readers that the function does not accept arguments, but is less visual clutter.